pkg/ingester/ingester_ring.go

```diff
@@ -73,10 +75,15 @@ func (cfg *RingConfig) Validate() error {
 		if cfg.TokensFilePath != "" {
 			return fmt.Errorf("%q token generation strategy requires %q to be empty", spreadMinimizingTokenGeneration, tokensFilePathFlag)
 		}
-		_, err := ring.NewSpreadMinimizingTokenGenerator(cfg.InstanceID, cfg.InstanceZone, cfg.SpreadMinimizingZones, nil)
+		_, err := ring.NewSpreadMinimizingTokenGenerator(cfg.InstanceID, cfg.InstanceZone, cfg.SpreadMinimizingZones, cfg.TokenGenerationCanJoinEnabled, nil)
 		return err
 	}
 
+	// at this point cfg.TokenGenerationStrategy must be randomTokenGeneration
+	if cfg.TokenGenerationCanJoinEnabled {
+		return fmt.Errorf("%q must be false in %q token generation strategy", tokenGenerationCanJoinEnabledFlag, randomTokenGeneration)
```

Comment:
[nit] Reading the actual strategy, just in case we'll support more in the future.

```suggestion
		return fmt.Errorf("%q must be false when using %q token generation strategy", tokenGenerationCanJoinEnabledFlag, cfg.TokenGenerationStrategy)
```